File: ruby/samples/stock_prices.rb
# Load the gem
require 'bundler/setup'
require 'intrinio-sdk'
require 'awesome_print'

# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR_API_KEY'
end

security_api = Intrinio::SecurityApi.new

identifier = "AAPL" # String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

opts = { 
  start_date: Date.today - 90, # Date | Return prices on or after the date
  end_date: Date.today, # Date | Return prices on or before the date
  frequency: "daily", # String | Return stock prices in the given frequency
  next_page: '' # String | Gets the next page of data from a previous API call
}

begin
  results = security_api.get_security_stock_prices(identifier, opts)
  results.stock_prices.each do |stock_price|
    ap stock_price
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling SecurityApi->get_security_stock_prices: #{e}"
end

File: javascript-node/samples/stock_prices.js
var intrinio = require('intrinio-sdk');
const util = require('util')

intrinio.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR_API_KEY";

var security_api = new intrinio.SecurityApi();

var identifier = "AAPL"; // String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

var opts = { 
  'startDate': new Date("2018-01-01"), // Date | Return prices on or after the date
  'endDate': new Date("2019-01-01"), // Date | Return prices on or before the date
  'frequency': "daily", // String | Return stock prices in the given frequency
  'nextPage': null // String | Gets the next page of data from a previous API call
};

security_api.getSecurityStockPrices(identifier, opts).then(function(data) {
  console.log(util.inspect(data, false, null, true));
}, function(error) {
  console.error(error);
});
